### PR TITLE
Progressive onboarding full page design

### LINKED ISSUE #4746: Full Page Design for Progressive Onboarding
### Description

This issue will cover implementing the **designs** for Progressive Onboarding. Implementation of the actual logic of the form will be covered by #5473, and implementation of the custom component elements will be covered by #4744, so this issue will mainly cover putting these two parts together.

### Acceptance Criteria

- PO flow matches the designs as shown in paJDYF-5IU-p2

### PR DESCRIPTION
**Pull request by ismaeldcom**
Fixes #4746

#### Changes proposed in this Pull Request
- Fix `CustomSelectControl` style issue on WP +6.1.
- Adjust style to match final design – sizes, fonts, margins...
- Update copies to match final design.
- Scroll to top on `Stepper` change.
- Add `BannerNotice` to `PersonalDetails`.
- Implement `Loading` step design.
- Make the onboarding flow full screen, using existing WC classes and `admin_body_class` filter.
- Update `Stepper` to implement progress, prevStep(), exit(), onStepChange(), and onExit().
- Move onboarding `Step` component to a new file, adding the missing nav bar with progress bar, back and close buttons, and WCPay logo.
- Make validation compatible with back functionality.
- Create `PhoneNumber` Field and OnboardingField.
- Persist phone country code between steps, to ensure the country doesn't change if the user goes back. Using a new `temp` property on the onboarding context.

##### Desktop
- With no account onboarded or WCPay Dev `Force the plugin to act as disconnected from WCPay` checked.
- Go to **Payments** and click on **Finish setup**
- The onboarding flow should be full screen, and load without showing WP Admin at all.
- Fill the first step and click on continue.
- The progress bar should grow as expected.
- Clicking on `Back` should move to the previous step, clicking it on the first step should redirect to the previous page in the history (Payments connect page in this case). The same for the `X` button in the top right corner.
- Complete the flow, every step should match the provided final design.

##### Mobile
- Check the flow on a mobile screen.
- Should work as expected without any visual issue.
- On step change the screen should scroll back to the top of the page.

#### Video

https://user-images.githubusercontent.com/7670276/232497422-8d944490-0c27-4237-a0fb-bdd556e293f9.mov

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6055` or branch name `add/4746-po-full-page-design` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/4746-po-full-page-design) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: aa8d7e9537b0438e785f5fade38df9b36c85d044
- Build time: 2023-04-28 10:50:57 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: changelog/add-4746-po-full-page-design [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Significance: patch
+Type: dev
+Comment: Behid progressive onboarding feature flag. Full page design.
```
Comment (dmallory42) [inline changelog/add-4746-po-full-page-design]: Nitpick: typo here in `behid` 🙂

**Review by dmallory42 — APPROVED** [review]
Looking awesome! 🎉 Thanks for working on this! 

I can't really find anything to nitpick - I really like the way the code is structured, very elegant and easy to understand. Everything works as expected upon testing and matches the design. I also tested using mobile emulator in the browser and it displayed well on various device screen sizes.

We should probably get a second pair of eyes from design, and then this looks good to go to me!

File: changelog/add-4746-po-full-page-design (hunk 1 above)
Comment (ismaeldcom) [reply changelog/add-4746-po-full-page-design]: Nice catch! Addressed in eac2dbe

**Comment by ismaeldcom** [comment]
>We should probably get a second pair of eyes from design, and then this looks good to go to me!

Thanks! For sure, I forgot to manually ping them 🙂 

@elizaan36 could you review it? If you want to see it on a JN site, remember to install WCPay Dev tools and uncheck both `Enable API request redirection` and `Proxy WPCOM requests`, and check `Enable Progressive Onboarding`.

**Comment by oaratovskyi** [comment]
Looks amazing 🔥

**Comment by dmallory42** [comment]
Noting here a small copy update I asked about in Figma (will share link in Slack) - would it be possible to update that as part of this PR please, @ismaeldcom? Thanks!

**Comment by elizaan36** [comment]
Thanks @ismaeldcom I went through the first few screens and looks great. A couple minor notes and one bigger one :) 

First, I got stuck on the "Where is your business legally registered?" The dropdown won't open for me. I'm testing on a JN site on Chrome.
Screen Shot 2023-04-18 at 4 55 54 PM

Visual notes
- The WooPayments logo looks off center to the right
- The "Back" button and X to close are a little too heavy in weight. The Back link should be Regular body text and X to close would match that weight.
- In the phone number field the country code dropdown isn't showing the blue background so there's an inconsistency with the rest of the field.
Will paste this screenshot to illustrate the visual notes and so you can see what I'm seeing.
Screen Shot 2023-04-18 at 4 55 14 PM

**Comment by ismaeldcom** [comment]
Hey @dmallory42! Copy addressed in 208ebf6 🙂 

Thanks @elizaan36 for the detailed review!

>I got stuck on the "Where is your business legally registered?" The dropdown won't open for me.

Country and business types/structures comes from the server, if it's empty somehow it's not loading them properly. We will adjust it in another issue, to preload them as soon as possible, to avoid showing a spinner there.

>The WooPayments logo looks off center to the right

Good catch! I missed that it's centered between the two buttons and not within the screen.

>The "Back" button and X to close are a little too heavy in weight. The Back link should be Regular body text and X to close would match that weight.

I copied the font weight directly from Figma, probably the issue is because of the different font used there. A regular weight doesn't match it either, but looks better with the updated icons. Sometimes it is not easy to pick the right one, because we have two different versions of Gridicons, plus WP ones.

>In the phone number field the country code dropdown isn't showing the blue background so there's an inconsistency with the rest of the field.

That's caused by the browser prefill functionality, it behaves differently based on every browser. But I updated it to place the country code within the input, to make it consistent.

This is how it looks now, let me know if you see any additional issue 🙂 

Screenshot 2023-04-19 at 12 23 46

**Comment by elizaan36** [comment]
Nice @ismaeldcom! Let me know when the rest of the flow after business location can be tested and I'll jump in there. Also there's a question on copy for one of the screens again, will share in Slack.

**Review by dmallory42 — APPROVED** [review]
Tested again after the latest round of changes addressing feedback. All looks great, thanks for working on this!

I made a few small comments in Slack (mentioning them here just for posterity), one about the height of the country select input (which is quite large and can cause the page layout to stretch) - this has already been addressed.

The other relates to using the new `Woo Payments` logo, and using `WooCommerce Payments` in the copy. We'll check with design, and this can be addressed in the new copy changes issue if needed.

Let's :shipit: !

**Pull request merged**